File: backend/src/main/java/com/app/mevocab/internal/quiz/dto/request/QuestionSaveRequest.java
package com.app.mevocab.internal.quiz.dto.request;

import com.app.mevocab.common.enums.QuestionType;
import com.app.mevocab.internal.quiz.dto.Question;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class QuestionSaveRequest {
    String quizID;
    List<Question> questions;
}

File: frontend/lib/data/models/dto/Request/QuestionSaveRequest.dart
import '../../Question.dart';

class QuestionSaveRequest {
  final String quizID;
  final List<Question> questions;

  QuestionSaveRequest({
    required this.quizID,
    required this.questions,
  });

  Map<String, dynamic> toJson() {
    return {
      'quizID': quizID,
      'questions': questions.map((q) => q.toJson()).toList(),
    };
  }
}
